streaming/stream_manager.hh

```diff
@@ -54,12 +54,7 @@ struct stream_bytes {
         ret += y;
         return ret;
     }
-    friend bool operator!=(const stream_bytes& x, const stream_bytes& y) {
-        return x.bytes_sent != y.bytes_sent && x.bytes_received != y.bytes_received;
```

Comment:
Yikes, this implementation was wrong? I should have been "||", not "&&"? Or am I missing something?

Reply:
ahh, it was wrong! but fortunately, this operator is never used. i just removed the defaulted `operator==` and the tree still compiles. was wondering if i should create an issue for it. guess i will instead create a follow-up PR to drop this comparison operator after this change lands.

Reply:
#13697